rename header to avoid clash.

shlomif opened the pull request:
See https://en.wikipedia.org/wiki/Assert.h .

shlomif committed:
Rename headers to avoid clash.

See e.g: https://en.wikipedia.org/wiki/Assert.h , which clashes with our
assert.h and other too generic, non namespaced-prefixed, header
basenames.

We have placed all header files under include/OB_table/ per the discussion at:
https://github.com/Veltas/OB_table/pull/1 .

Veltas commented:
Thanks for the PR, and thanks for trying out the code!

One issue: this PR breaks the build.

There *shouldn't* be an issue with sharing the assert.h name, the 'right way' to include these headers would be to put them in a directory e.g. `OB_table/` and then `"assert.h"` in the header will refer to the header in the same directory, since it is not angle brackets it will not search for the system one. If someone wants to use these files they should put them in e.g. `/usr/local/include/OB_table` directory or named project sub-directory to avoid name collisions.

*However*, since you ran into an issue using it as-is, it does seem to make it harder for people to use these files, so I shall rename it as you suggest to make it just a little bit smoother for people.

If you want you can submit another PR, please do `make clean` and `make tests` first to make sure it is not broken. If you don't or want me to I will make a change myself later, because I think it will improve the code.

shlomif commented:
@Veltas : thanks for reviewing my changes! I've now fixed the test suite and took the initiative to rename other possibly also generic headers and rebased. One note is that we may wish to use the approach I took in [rinutils](https://github.com/shlomif/rinutils/) and place all headers under a sub directory - e.g `#include <OB_table/assert.h>`.

Veltas commented:
Yes @shlomif that sounds like the preferred way, could you rearrange the files with include/OB_table/ and source/ directories instead of renaming them? Or if you don't want to I can do it.

shlomif commented:
On Mon, 20 Apr 2020 10:20:57 -0700
Christopher Leonard <notifications@github.com> wrote:

> Yes @shlomif that sounds like the preferred way, could you rearrange the
> files with include/OB_table/ and source/ directories instead of renaming
> them? Or if you don't want to I can do it.
> 

OK, I'll do it. Thanks!

-- 

Shlomi Fish       https://www.shlomifish.org/
https://github.com/shlomif/validate-your-html - Validate Your HTML

If the mountain will not come to Muhammad, then Muhammad will go to the
mountain.  — Francis Bacon

Please reply to list if it's a mailing list post - https://shlom.in/reply .

shlomif commented:
@Veltas : done now - please let me know what you think.

Veltas commented on include/OB_table/assert.h:
```diff
@@ -1,7 +1,7 @@
 #define VV_ASSERT_INCLUDED
 
-#include "util.h"
+#include "OB_table/util.h"
```
Hi, lines like this should either change to `<OB_table/util.h>` or stay the same, because `""` is for local lookup and `<>` is for using the -I and system include paths.

shlomif replied:
@Veltas : changed now and "make tests" passes. Thanks!

Veltas commented:
Looks good, thanks
